stumpy/stumpi.py

```diff
@@ -471,6 +453,61 @@ def _update(self, t):
         self._Σ_T = Σ_T_new
         self._T_subseq_isconstant = T_subseq_isconstant_new
 
+    @staticmethod
+    @njit
+    def _update_egress_PI(D, P, I, excl_zone, n_appended=0):
+        """
+        Given the 1D array distance profile, `D`, of the last subsequence of T,
+        update (in-place) the (top-k) matrix profile, `P`, and the matrix profile
+        index, I, of T.
+
+        Parameters
+        ----------
+        D : numpy.ndarray
+            A 1D array (with dtype float) representing the distance profile of
+            the last subsequence of T
+
+        P : numpy.ndarray
+            A 2D array representing the matrix profile of T,
+            with shape (len(T) - m + 1, k)
+
+        I : numpy.ndarray
+            A 2D array representing the matrix profile index of T,
+            with shape (len(T) - m + 1, k)
+
+        excl_zone : int
+            Size of the exclusion zone. That is, after finding the next-best-match
+            located at index `idx`, we ignore subsequences with start index in range
+            (idx -  excl_zone, idx + excl_zone + 1).
+
+        n_appended : int
+            Number of times the timeseries start point is shifted one to the right
+
+        Returns
+        -------
+        None
+        """
+        core._apply_exclusion_zone(D, D.shape[0] - 1, excl_zone, np.inf)
```

Comment:
**Note:** I decided to include this in the function. Otherwise, one needs to make sure the elements corresponding to the trivial subsequences of last subsequences in `D` are set to `np.inf`

Reply:
It's probably okay